src/Octoshift/Services/BbsApi.cs

```diff
@@ -73,11 +76,76 @@ public virtual async Task<IEnumerable<(int Id, string Key, string Name)>> GetPro
             .ToListAsync();
     }
 
+    public virtual async Task<(int Id, string Key, string Name)> GetProject(string projectKey)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}";
+        var response = await _client.GetAsync(url);
+
+        var project = JObject.Parse(response);
+        return ((int)project["id"], (string)project["key"], (string)project["name"]);
+    }
+
     public virtual async Task<IEnumerable<(int Id, string Slug, string Name)>> GetRepos(string projectKey)
     {
         var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos";
         return await _client.GetAllAsync(url)
             .Select(x => ((int)x["id"], (string)x["slug"], (string)x["name"]))
             .ToListAsync();
     }
+
+    public virtual async Task<bool> GetIsRepositoryArchived(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}?fields=archived";
+        var response = await _client.GetAsync(url);
+
+        var data = JObject.Parse(response);
+        return (bool)data["archived"];
+    }
+
+    public virtual async Task<IEnumerable<(int Id, string Name)>> GetRepositoryPullRequests(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}/pull-requests?state=all";
+        return await _client.GetAllAsync(url)
+            .Select(x => ((int)x["id"], (string)x["name"]))
+            .ToListAsync();
+    }
+
+    public virtual async Task<DateTime> GetRepositoryLatestCommitDate(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}/commits?limit=1";
+
+        try
+        {
+            var response = await _client.GetAsync(url);
+
+            var commit = JObject.Parse(response);
+
+            var authorTimestamp = 0L;
+            if (commit?["values"] != null && commit["values"].Any())
+            {
+                authorTimestamp = (long)commit["values"][0]["authorTimestamp"];
+            }
+
+            var dateTime = authorTimestamp > 0 ? DateTimeOffset.FromUnixTimeMilliseconds(authorTimestamp).DateTime : DateTime.MinValue;
+
+            return dateTime.Date;
+        }
+        catch (HttpRequestException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
+        {
+            return DateTime.MinValue;
+        }
+    }
```

Comment:
It looks like we're returning `DateTime.MinValue` from this method when a latest commit is not found.  Perhaps we should return `null` instead?

Additionally, in this implementation, it looks like we're relying on `(long)commit["values"][0]["authorTimestamp"` to not be able to cast an expected input to make a decision on returning a "missing commit" value or not.  Perhaps we could inspect `commit["values"][0]["authorTimestamp"]` to see if it contains number characters instead?